bitcoin/src/blockdata/script/builder.rs

```diff
@@ -54,6 +57,23 @@ impl Builder {
         self.push_slice(&<&PushBytes>::from(&buf)[..len])
     }
```

Comment:
Perhaps this function is cleaner if written like this?

```rust
    /// Adds instructions to push an integer onto the stack.
    ///
    /// Integers are encoded as little-endian signed-magnitude numbers, but there are dedicated
    /// opcodes to push some small integers.
    ///
    /// # Errors
    ///
    /// Only errors if `data == i32::MIN` (CScriptNum cannot have value -2^31).
    pub fn push_int(self, data: i32) -> Result<Builder, Error> {
        if data == i32::MIN {
            Err(Error::NumericOverflow)
        }

        else if data == -1 {
            Ok(sef.push_opcode(OP_PUSHNUM_NEG1))
        } else if data >= 0 {
            Ok(self.push_uint(data as u32))
        } else {
            Ok(self.push_int_non_minimal(data.into()))
        }
    }
```

Its up to you but I'd personally order the functions `push_int`, `push_uint`, and then `push_int_non_minimal`.

Also if we are touching these functions perhaps we should re-name the parameter, `data` does not convey much info.  `n` is the name that jumps out at me but some folk don't like short names, is there any other conventional name we could use @apoelstra?

Reply:
That's a great retouch! I will reflect it when the new name of param is ready.